packages/models-library/src/models_library/api_schemas_webserver/catalog.py

```diff
@@ -133,3 +135,52 @@ class Config(InputSchema.Config):
 class ServiceResourcesGet(api_schemas_catalog_services.ServiceResourcesGet):
     class Config(OutputSchema.Config):
         ...
+
+
+class DEVServiceGet(ServiceGet):
+    # pylint: disable=too-many-ancestors
+
+    history: list[ServiceRelease] = Field(
+        default=[],
+        description="history of releases for this service at this point in time, starting from the newest to the oldest."
+        " It includes current release.",
+    )
+
+    class Config(OutputSchema.Config):
+        schema_extra: ClassVar[dict[str, Any]] = {
+            "example": {
+                **_EXAMPLE,  # 1.0.0
+                "history": [
+                    {
+                        "version": "1.0.5",
+                        "version_display": "Summer Release",
+                        "release_date": "2024-07-20T15:00:00",
+                    },
+                    {
+                        "version": _EXAMPLE["version"],
+                        "compatibility": {
+                            "can_update_to": "1.0.5",
+                        },
+                    },
+                    {"version": "0.9.11"},
+                    {"version": "0.9.10"},
+                    {
+                        "version": "0.9.8",
+                        "compatibility": {
+                            "can_update_to": "0.9.10",
```

Comment:
I understand the idea with this but how will you fill this information?
- I guess when someone releases a service he/she knows what service it can patch,
- is this can_update_to not a list? 
- how do you handle service key changes?

Reply:
I am also wondering about how you will do this. If this information is "static" (i.e. in some service metadata somewhere and not something you compute on the fly) then it would probably be easier to do it the other way around. I.e. add a `can_replace` field in version 0.9.10 so that you don't have to update the data of version 0.9.8 by the time 0.9.10 is released.

Reply:
The compatibility is evaluated on the backend on the following rules:
- implicit version semantics rules (e.g. 3.2.4 can be updated to 3.4.10)
- same inputs/outputs (e.g. if version semantics makes it compatible we double check that i/o did not change)
- explicit user compatibility rule (e.g. user enforces that v3.4 is compatible with v4 even if version semantics does not allow it)

Regarding `can_update_to`, I follow what the front-end needs. For the moment @odeimaiz upgrades to the "latest compatible" and also shows the latest version. For that reason `can_update_to` is just one version and not a list

![image](https://github.com/ITISFoundation/osparc-simcore/assets/32402063/60f3c3e6-4dc7-453e-9f9f-5e4b8a9fcca0)

Reply:
> I am also wondering about how you will do this. If this information is "static" (i.e. in some service metadata somewhere and not something you compute on the fly) then it would probably be easier to do it the other way around. I.e. add a `can_replace` field in version 0.9.10 so that you don't have to update the data of version 0.9.8 by the time 0.9.10 is released.

@bisgaard-itis First, let me point out that in this PR I also clearly separate the "static" part (`service_metadata_published`) from the "editable" (`service_metadata_editable`)part. SEE
![image](https://github.com/ITISFoundation/osparc-simcore/assets/32402063/aa42b433-ce4d-4596-a524-d540d00cf11c)
The way I see is that they work as docker-compose, i.e. the editable part overrides the published.

I try to provide the front-end with exactly what is needed. we can talk offline to see your idea since I do not totally understand your point.